test/test_helpers.jl

```diff
@@ -1,14 +1,14 @@
 """
     @test_modern
 
-Use this to mark tests that should pass on modern (>=v1.1) versions of julia,
+Use this to mark tests that should pass on modern (>=v1.4) versions of julia,
 but should be marked as `@test_broken` on Julia 1.0.
 For more complicated cases, or where it shouldn't be defined at all on julia versions,
 you can use `if VERSION` directly.
 """
 macro test_modern(expr...)
     if VERSION >= v"1.4"
-        return :(@test $(expr...))
+        return :(@test $(expr...)) |> esc
```

Comment:
Would
```suggestion
        return :(@test $(esc(expr...)))
```
also work? I personally find it clearer than putting everything in an `esc`

If there is a reason to have the whole thing in `esc`, should we change it to
```suggestion
        return esc(:(@test $(expr...)))
```
to be consistent with the rest of the package style?

Reply:
No strong feelings, was just trying to make the minimal change to make it work. Have you checked that the suggestion does in fact work, including on 1.6? CI won't (yet) check this I don't think.

Reply:
I've tested on the 1.6 beta that is downloadable from the website but have not built from source to include the bugfix. Both ways solve the macro problem (but the crash is still there)

Reply:
Ok! Am just a bit concerned by that splat, since there's no 2-arg method:
```
julia> esc(1,2)
ERROR: MethodError: no method matching esc(::Int64, ::Int64)
Closest candidates are:
  esc(::Any) at essentials.jl:482
```

Reply:
Ah great catch! This shouldn't happen (as `@test` only takes one thing too) but if it does the message is better with the current choice as `@test` would fail, not `esc`. I wonder if instead we should restrict the macro to `macro test_modern(expr)` in the first place?

Reply:
Indeed, `@test 1 ≈ 1.1 atol=0.2` is in fact one expression, so perhaps you can just delete the `...`.

Reply:
Is it really one expression? How about this example?
```
julia> macro mymacro(expr)
           @show expr
       end
@mymacro (macro with 1 method)

julia> @mymacro3 1≈1.1 atol=0.2
ERROR: LoadError: MethodError: no method matching @mymacro3(::LineNumberNode, ::Module, ::Expr, ::Expr)
```
The `@test` implementation actually seems to take an arbitrary number of kwargs
https://github.com/JuliaLang/julia/blob/5f10eb9085fd5a622430d020c8bc9141c7acb7c7/stdlib/Test/src/Test.jl#L372

I think we have found a reason why the `esc` should be on the outside as your originally suggest.

Reply:
Oh I'm sorry, I tested the wrong thing:
```
julia> @pr @test 1 ≈ 1.1 atol=0.2
ex1 = :(#= REPL[312]:1 =# @test 1 ≈ 1.1 atol = 0.2)
ex2 = nothing
```
But this is a drop-in for `@test`, not a wrapper. So it does need 2 args.

Reply:
I forget, do you have merge rights?

Reply:
I do, but not tagging rights, I believe.

Reply:
I don't think we need to bump a version because the PR doesn't change any code in `src`

Reply:
I think the version is already bumped one ahead of what's tagged, because of an earlier PR. 

Would be good to tag a version which passes on 1.6- though, for the sake of the mass-testing of registered packages which gets done before release. (That's how I saw the bug in #152 in the first place.)

Reply:
oh ok, that makes sense